docker-compose.yml

```diff
@@ -0,0 +1,59 @@
+# Consul is the service catalog that helps discovery between the components
+# Change "-bootstrap" to "-bootstrap-expect 3", then scale to 3 or more to
+# turn this into an HA Consul raft.
+consul:
+  image: autopilotpattern/consul:latest
+  command: >
+    /usr/local/bin/containerpilot
+    /bin/consul agent -server
+    -bootstrap
+    -config-dir=/etc/consul
+    -ui-dir /ui
+  restart: always
+  mem_limit: 128m
+  env_file: _env
+  expose:
+    - 53
+    - 8300
+    - 8301
+    - 8302
+    - 8400
+    - 8500
+    - 9090
+  dns:
+    - 127.0.0.1
+  labels:
+    - triton.cns.services=docker-registry-consul
+    - com.docker.swarm.affinities=["container!=~*"]
+
+# Our Docker registry
+registry:
+  image: autopilotpattern/docker-registry:latest
+  restart: always
+  mem_limit: 128m
```

Comment:
Have we tested this with larger image layers? I'm worried large image layers don't end up getting streamed in-memory without giant chunks because, well, I don't trust upstream not to foul that up. 😀